src/index.ts

```diff
@@ -1067,20 +1095,26 @@ export function createForgoInstance(customEnv: any) {
     nodes: ArrayLike<ChildNode>,
     from: number,
     to: number
-  ) {
+  ): ChildNode[] {
+    const justDeletedNodes: ChildNode[] = [];
+
     const nodesToRemove = sliceNodes(nodes, from, to);
     if (nodesToRemove.length) {
       const parentElement = nodesToRemove[0].parentElement as HTMLElement;
       const deletedNodes = getDeletedNodes(parentElement);
       for (const node of nodesToRemove) {
         // If the consuming application has manually mucked with the DOM don't
         // remove things it added
-        if (!getForgoState(node)) continue;
+        const state = getForgoState(node);
+        if (!state) continue;
 
         node.remove();
+        justDeletedNodes.push(node);
```

Comment:
Under what circumstances do we expect `deletedNodes` and `justDeletedNodes` to diverge? Could we add a comment explaining why we need to track both?

Reply:
`justDeletedNodes` were just deleted, and specific to the element currently being considered. `deletedNodes` is the entire set of marked-for-deletion nodes on a parent, which might have happened due to the deletion of sibling nodes in addition to the currently considered node.

In the code, `deletedNodes` is a smaller subset of `sliceNodes(nodes, from, to)` - where from and to define the search range for finding a matching component.

But this is going away though, due to the other change you proposed.